frame/pablo/src/lib.rs

```diff
@@ -0,0 +1,782 @@
+		/// Emits `PoolCreated` event when successful.
+		// TODO: enable weight
+		#[pallet::weight(10_000)]
+		pub fn create(
+			_origin: OriginFor<T>,
+			_pool: PoolConfiguration<T::AccountId, T::AssetId>,
+		) -> DispatchResult {
+			Ok(())
+		}
+
+		/// Execute a buy order on pool.
+		///
+		/// Emits `Swapped` event when successful.
+		#[pallet::weight(10_000)]
+		pub fn buy(
+			_origin: OriginFor<T>,
+			_pool_id: T::PoolId,
+			_asset_id: T::AssetId,
+			_amount: T::Balance,
+			_keep_alive: bool,
+		) -> DispatchResult {
+			Ok(())
+		}
+
+		/// Execute a sell order on pool.
+		///
+		/// Emits `Swapped` event when successful.
+		#[pallet::weight(10_000)]
+		pub fn sell(
+			_origin: OriginFor<T>,
+			_pool_id: T::PoolId,
+			_asset_id: T::AssetId,
+			_amount: T::Balance,
+			_keep_alive: bool,
+		) -> DispatchResult {
+			Ok(())
+		}
+
+		/// Execute a specific swap operation.
+		///
+		/// The `quote_amount` is always the quote asset amount (A/B => B), (B/A => A).
+		///
+		/// Emits `Swapped` event when successful.
+		#[pallet::weight(10_000)]
+		pub fn swap(
+			_origin: OriginFor<T>,
+			_pool_id: T::PoolId,
+			_pair: CurrencyPair<T::AssetId>,
+			_quote_amount: T::Balance,
+			_min_receive: T::Balance,
+			_keep_alive: bool,
+		) -> DispatchResult {
+			Ok(())
+		}
+
+		/// Add liquidity to a stable-swap pool.
+		///
+		/// Emits `LiquidityAdded` event when successful.
+		#[pallet::weight(10_000)]
+		pub fn add_liquidity(
+			_origin: OriginFor<T>,
+			_pool_id: T::PoolId,
+			_base_amount: T::Balance,
+			_quote_amount: T::Balance,
+			_min_mint_amount: T::Balance,
+			_keep_alive: bool,
+		) -> DispatchResult {
+			Ok(())
+		}
+
+		/// Remove liquidity from stable-swap pool.
+		///
+		/// Emits `LiquidityRemoved` event when successful.
+		#[pallet::weight(10_000)]
+		pub fn remove_liquidity(
+			_origin: OriginFor<T>,
+			_pool_id: T::PoolId,
+			_lp_amount: T::Balance,
+			_min_base_amount: T::Balance,
+			_min_quote_amount: T::Balance,
+		) -> DispatchResult {
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		pub(crate) fn do_create_pool(pool: PoolOf<T>) -> Result<T::PoolId, DispatchError> {
+			let pool_id =
+				PoolCount::<T>::try_mutate(|pool_count| -> Result<T::PoolId, DispatchError> {
+					let pool_id = *pool_count;
+					Pools::<T>::insert(pool_id, pool.clone());
+					*pool_count = pool_id.safe_add(&T::PoolId::one())?;
+					Ok(pool_id)
+				})?;
+			match pool {
+				PoolConfiguration::StableSwap(stable_swap_pool_info) => {
+					let pool = stable_swap_pool_info;
+					Self::deposit_event(Event::PoolCreated { pool_id, owner: pool.owner.clone() });
+					Ok(pool_id)
+				},
+			}
+		}
+
+		pub(crate) fn get_pool(pool_id: T::PoolId) -> Result<PoolOf<T>, DispatchError> {
+			Pools::<T>::get(pool_id).ok_or_else(|| Error::<T>::PoolNotFound.into())
+		}
+
+		pub(crate) fn account_id(pool_id: &T::PoolId) -> T::AccountId {
+			T::PalletId::get().into_sub_account(pool_id)
+		}
+
+		fn get_exchange_value_for_stable_swap(
```

Comment:
We could also add this StableSwap specific code starting from here in a later PR with tests etc., as now this does not really serve any purpose.